### ISSUE TITLE
Garbage on embedded DNS port crashes daemon

### ISSUE THREAD

**Issue by Anonymooseable** [docker/docker#20350]
I'm running debian stable.
`uname -a`:
```
Linux computer 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt20-1+deb8u3 (2016-01-17) x86_64 GNU/Linux
```
`docker version`:
```
Client:
 Version:      1.10.1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   9e83765
 Built:        Thu Feb 11 19:09:42 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   9e83765
 Built:        Thu Feb 11 19:09:42 2016
 OS/Arch:      linux/amd64
```
`docker info`:
```
Containers: 11
 Running: 0
 Paused: 0
 Stopped: 11
Images: 169
Server Version: 1.10.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 202
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.62 GiB
Name: computer
ID: DZVQ:A2VP:KUGX:AGI5:RFIA:JDHO:HHB6:JACK:6QXK:4YRS:B6NP:VTU2
```

To reproduce:
```
$ docker network create crash
$ docker run -ti --net crash debian bash
container# apt-get update && apt-get -y install netcat-traditional
container# echo "hello" | nc -u 127.0.0.11 53
```
Expected results: daemon doesn't crash
Actual results: daemon crashes

**Comment by thaJeztah** [comment]
I was able to reproduce this, thanks for reporting!

**Comment by tiborvass** [comment]
Reopening because it was fixed in master but not for 1.10.2

**Comment by tiborvass** [comment]
Fixed by #20510
